src/ui/public/agg_types/metrics/derivative.js

```diff
@@ -0,0 +1,142 @@
+import AggTypesMetricsMetricAggTypeProvider from 'ui/agg_types/metrics/metric_agg_type';
+import orderAggTemplate from 'ui/agg_types/controls/sub_agg.html';
+import _ from 'lodash';
+import $ from 'jquery';
+import VisAggConfigProvider from 'ui/vis/agg_config';
+import VisSchemasProvider from 'ui/vis/schemas';
+
+export default function AggTypeMetricDerivativeProvider(Private) {
+  const DerivativeAggType = Private(AggTypesMetricsMetricAggTypeProvider);
+  const AggConfig = Private(VisAggConfigProvider);
+  const Schemas = Private(VisSchemasProvider);
+
+  const aggFilter = ['!top_hits', '!percentiles', '!percentile_ranks', '!median', '!std_dev'];
+  const orderAggSchema = (new Schemas([
+    {
+      group: 'none',
+      name: 'orderAgg',
+      title: 'Order Agg',
+      aggFilter: aggFilter
+    }
+  ])).all[0];
+
+  return new DerivativeAggType({
+    name: 'derivative',
+    title: 'Derivative',
+    makeLabel: function (aggConfig) {
```

Comment:
I suppose I disagree on the second point, I think this file would be more readable with less LOC.  The more focused functions are, the more I can concentrate on what it is doing, and switch to other focus areas as a follow up.  This file has a whole controller in it, which could also probably be split out.  It also becomes tough to follow with a lot of indentation.

 For instance, the creation of the params argument is a bit confusing for me to follow.  It looks like the object parameters are later translated into classes based on the type and/or name field of the object.  I got a little lost however because there are two agg_params.js files.  More functions generally mean more comments, or barring that, self documenting code using the function name.

I also look at this file and wonder if we can separate the business logic from the angular portions.  makeLabel has nothing angular specific about it, but this file has a reference to $scope and Private, which is angular specific.

Anyway, I'm fine if you leave as is, just expressing my opinion.  This is the first I've looked at the visualize code, so trying to follow the logic is a bit tough.